client/blocks/inline-help/admin-sections.js

```diff
@@ -452,7 +451,9 @@ export function filterListBySearchTerm( searchTerm = '', collection = [], limit
 			partialMatches.push( item );
 		} else if (
 			'en' === getLocaleSlug() &&
-			intersection( item.synonyms, searchTermWords ).length > 0
+			[ item.synonyms ?? [], searchTermWords ].reduce( ( a, b ) =>
+				a.filter( ( c ) => b.includes( c ) )
+			).length > 0
```

Comment:
Wow this looks like a puzzle 🙂 With an array of two items, the reducer will ever be called only once, and we can unroll the `reduce` call to:
```js
item.synonyms.filter( ( c ) => searchTermWords.includes( c ) ).length > 0;
```
which can be further converted to:
```js
item.synonyms.some( ( c ) => searchTermWords.includes( c ) )
```